File: MaximumSubarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum = nums[0];
        int max = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (sum + nums[i] > nums[i]) {
                sum = sum +nums[i];
            } else {
                sum = nums[i];
            }
            
            if (sum > max) {
                max = sum;
            }
        }
        return max;
    }
}

File: MaximumSubarray.scala
object Solution {
    def maxSubArray(nums: Array[Int]): Int = {
        var max = nums(0);
        var sum = nums(0);
        for (i <- 1 until nums.length) {
            sum = scala.math.max(sum+nums(i),nums(i));
            max = scala.math.max(sum,max);
        }
        max;
    }
}
